RELEASE_NOTES.md

```diff
@@ -85,6 +85,9 @@ Notable changes include:
         launch per SM. This resulted in a performance improvement for an
         application use case. Future work will expand this concept to other GPU
         kernel execution methods in RAJA.
+      * RAJA Modules is deprecated and no longer uses the "-fmodules" flag
+        since it hasnt been kept up to date. The RAJA_ENABLE_MODULES option
+        has been removed.
```

Comment:
How does this sound?

Reply:
Maybe replace "since it hasn't been kept up to date" with "since it can cause issues". Also, change "has been removed" to " will be removed in the next release". When something gets deprecated, we typically make users aware that it will be removed in the future.

Reply:
Sounds good! Thanks for edits

Reply:
I just wanted to clarify one thing I think got lost in the email thread - the ENABLE_MODULES option has been broken for a long time and doesn't enable modules. The issue was that the code block to enable the flag was looking for the `RAJA_ENABLE_MODULES` variable, which didn't exist. This PR renamed the ENABLE_MODULES option and caused this block to be executed, where it didn't used to be.

So there's not really anything to "remove" in terms of the option because it never worked.

Reply:
So @davidbeckingsale, you're advocating for just ripping it out? I guess that makes sense since if it didn't work, folks weren't using it, right?

Reply:
Yeah. The only way to enable it is to manually add the `-fmodules` flag. The deprecation will come from removing the modulemap file which we can do next release

Reply:
@kab163 will you please add an issue to document in the user guide how to enable the modules for CMake and regular make based build systems? We will add for the next release. Thank you.

Reply:
Sure thing. I'll update this deprecation message too